rosbag2_storage/include/rosbag2_storage/storage_interfaces/read_only_interface.hpp

```diff
@@ -28,7 +27,7 @@ namespace storage_interfaces
 {
 
 class ROSBAG2_STORAGE_PUBLIC ReadOnlyInterface
-  : public BaseInfoInterface, public BaseIOInterface, public BaseReadInterface
```

Comment:
I don't understand this change, or maybe just don't agree with it.

When I open a bag file read only, I shall still have the opportunity to read the meta data info, right? How would I do this with this current change?

Reply:
Here is our rationale:

- Reading the metadata yaml file should always be possible, even without opening the storage (e.g. when the storage is not known in advance) - and indeed, the storage should not be opened if it is not necessary.
- The logic to read the yaml file is the same for every plugin hence a storage plugin author does not need to implement it
- Therefore, it seems logical to expose this functionality outside of the storage_interface hierarchy.
Note that the information we get from the storage plugin itself is necessarily incomplete - it must be enriched with additional information from the filesystem.

It is of course possible to expose a non-abstract method inside the storage interface which just calls the the outside method. I'm just not sure I see the benefit.

In other words, with the current implementation, you can open your storage read-only (or read-write) and you can (in a separate call) query the information from the metadata file without having a storage handle.

Reply:
That assumption only holds when bag files are written with rosbag2. When, however, no yaml file is specified for let's say ros1 bags, then how do I get this type of info?

Reply:
Also, from an interface point of view. If I get a `ReadWriteInterface`, I would like to be able to call `get_info()` or `get_metadata()` on that instance. I don't see how I would do that right now, because with this change I have to rely on the plugin that this function is available.

Reply:
I see your first point. That's a part where we do indeed need a function to read the information from the storage directly (for read write, this is possible).